libcudacxx/test/libcudacxx/cuda/numeric/overflow.arithmetic/div_overflow.pass.cpp

```diff
@@ -0,0 +1,188 @@
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cuda/numeric>
+#include <cuda/std/cassert>
+#include <cuda/std/cstdint>
+#include <cuda/std/limits>
+#include <cuda/std/type_traits>
+#include <cuda/std/utility>
+
+#include "test_macros.h"
+
+template <typename Result, typename Lhs, typename Rhs>
+__host__ __device__ constexpr void
+test_div_overflow(const Lhs lhs, const Rhs rhs, bool overflow, bool special_case = false, Result expected = {})
+{
+  // test overflow_result<Result> div_overflow(Lhs lhs, Rhs rhs) overload
+  //   skip_special_case is used to skip special cases that are not valid in a constexpr context, e.g. INT_MIN / -1
+  //   but the result fits in the Result type
+  {
+    const auto result = cuda::div_overflow<Result>(lhs, rhs);
+    if (!overflow)
+    {
+      if (special_case)
+      {
+        assert(result.value == expected);
+      }
+      else
+      {
+        assert(result.value == static_cast<Result>(static_cast<Result>(lhs) / static_cast<Result>(rhs)));
+      }
+    }
+    assert(result.overflow == overflow);
+  }
+  // test bool div_overflow(Result& result, Lhs lhs, Rhs rhs) overload
+  {
+    Result result{};
+    const bool has_overflow = cuda::div_overflow<Result>(result, lhs, rhs);
+    if (!overflow)
+    {
+      if (special_case)
+      {
+        assert(result == expected);
+      }
+      else
+      {
+        assert(result == static_cast<Result>(static_cast<Result>(lhs) / static_cast<Result>(rhs)));
+      }
+    }
+    assert(has_overflow == overflow);
+  }
+}
+
+template <typename Lhs, typename Rhs, typename Result>
+__host__ __device__ constexpr void test_type()
+{
+  using cuda::std::is_same_v;
+  using cuda::std::is_signed_v;
+  using cuda::std::is_unsigned_v;
+  static_assert(is_same_v<decltype(cuda::div_overflow<Result>(Lhs{}, Rhs{})), cuda::overflow_result<Result>>);
+  [[maybe_unused]] constexpr auto lhs_min     = cuda::std::numeric_limits<Lhs>::min();
+  [[maybe_unused]] constexpr auto lhs_max     = cuda::std::numeric_limits<Lhs>::max();
+  [[maybe_unused]] constexpr auto result_max  = cuda::std::numeric_limits<Result>::max();
+  [[maybe_unused]] constexpr auto result_min  = cuda::std::numeric_limits<Result>::min();
+  [[maybe_unused]] constexpr auto neg_lhs_min = cuda::uabs(lhs_min);
+  //--------------------------------------------------------------------------------------------------------------------
+  //  trivial cases
+  // 1. 0 / 1 -> should not overflow
+  test_div_overflow<Result>(Lhs{0}, Rhs{1}, false);
+
+  // 2. 1 / 1 -> should not overflow
+  test_div_overflow<Result>(Lhs{1}, Rhs{1}, false);
+
+  // nvcc 12.0 doesn't support this case in a constexpr context
+#if _CCCL_CUDA_COMPILER(NVCC, !=, 12, 0)
```

Comment:
I believe we should then wrap the whole test in a `!__cccl_default_is_constant_evaluated()` so that we at least test runtime with 12.0